File: java-hangman-app/src/test/java/ru/polina_project/hangman/util/HangmanStageTest.java
package ru.polina_project.hangman.util;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class HangmanStageTest {
    @Test
    void testHangmanStages() {
        final String[] expectedStages = {
                """
                        +-----+
                        |     |
                        O     |
                       /|\\    |
                       / \\    |
                              |
                        =========
            """,
                """
                        +-----+
                        |     |
                        0     |
                       /|\\    |
                         \\    |
                              |
                        =========
            """,
                """
                        +-----+
                        |     |
                        0     |
                       /|\\    |
                              |
                              |
                        =========
            
            """,
                """
                        +-----+
                        |     |
                        O     |
                        |\\    |
                              |
                              |
                        =========
            """,
                """
                        +-----+
                        |     |
                        O     |
                        |     |
                              |
                              |
                        =========
            
            """,
                """
                        +-----+
                        |     |
                        O     |
                              |
                              |
                              |
                        =========
            """,
                """
                        +-----+
                        |     |
                              |
                              |
                              |
                              |
                        =========
            """,
                """
                              +
                              |
                              |
                              |
                              |
                              |
                        =========
            """
        };

        assertArrayEquals(expectedStages, HangmanStage.STAGES);
    }
}

File: kotlin-hangman-app/src/test/kotlin/ru/polina_project/hangman/util/HangmanStageTest.kt
package ru.polina_project.hangman.util

import kotlin.test.Test
import kotlin.test.assertEquals
class HangmanStageTest {


    @Test
    fun testStagesInitialization() {
        val expectedStageCount = 8

        // Проверка количества этапов
        assertEquals(expectedStageCount, STAGES.size)

        // Проверка первого этапа
        val expectedFirstStage =

            """
                        +-----+
                        |     |
                        O     |
                       /|\    |
                       / \    |
                              |
                        =========
            
            """
        assertEquals(expectedFirstStage, STAGES[0])

        // Проверка последнего этапа
        val expectedLastStage =
            """
                              +
                              |
                              |
                              |
                              |
                              |
                        =========
            
            """
        assertEquals(expectedLastStage, STAGES[STAGES.size - 1])
    }
}